var/spack/repos/builtin/packages/rocm-openmp-extras/package.py

```diff
@@ -0,0 +1,337 @@
+            libomptarget.format(src) + '/plugins/hsa/CMakeLists.txt')
+
+        filter_file(
+            r'rpath,\${LLVM_LIBDIR}\${OPENMP_LIBDIR_SUFFIX}',
+            'rpath,${LLVM_LIBDIR}${OPENMP_LIBDIR_SUFFIX}' +
+            '-Wl,-rpath,${COMGR_LIB}',
+            libomptarget.format(src) + '/plugins/hsa/CMakeLists.txt')
+
+        filter_file(
+            'ADDITIONAL_VERSIONS 2.7', 'ADDITIONAL_VERSIONS 3',
+            flang.format(src) + 'CMakeLists.txt')
+
+    def install(self, spec, prefix):
+        src = self.stage.source_path
+        gfx_list = "gfx700;gfx701;gfx801;gfx803;gfx900;gfx902;gfx906;gfx908"
+        openmp_extras_prefix = self.spec['rocm-openmp-extras'].prefix
+        devlibs_prefix = self.spec['rocm-device-libs'].prefix
+        devlibs_src = '{0}/rocm-openmp-extras/rocm-device-libs'.format(src)
+        hsa_prefix = self.spec['hsa-rocr-dev'].prefix
+        hsakmt_prefix = self.spec['hsakmt-roct'].prefix
+        comgr_prefix = self.spec['comgr'].prefix
+        llvm_inc = '/rocm-openmp-extras/llvm-project/llvm/include'
+        llvm_prefix = self.spec['llvm-amdgpu'].prefix
+        omp_bin_dir = '{0}/bin'.format(openmp_extras_prefix)
+        omp_lib_dir = '{0}/lib'.format(openmp_extras_prefix)
+        bin_dir = '{0}/bin'.format(llvm_prefix)
+        lib_dir = '{0}/lib'.format(llvm_prefix)
+
+        # flang1 and flang2 symlink needed for build of flang-runtime
+        # libdevice symlink to rocm-openmp-extras for runtime
+        if not (os.path.islink((os.path.join(bin_dir, 'flang1')))):
+            os.symlink(os.path.join(omp_bin_dir, 'flang1'),
+                       os.path.join(bin_dir, 'flang1'))
+        if not (os.path.islink((os.path.join(bin_dir, 'flang2')))):
+            os.symlink(os.path.join(omp_bin_dir, 'flang2'),
+                       os.path.join(bin_dir, 'flang2'))
+        if not (os.path.islink((os.path.join(lib_dir, 'libdevice')))):
+            os.symlink(os.path.join(omp_lib_dir, 'libdevice'),
+                       os.path.join(lib_dir, 'libdevice'))
+
+        # Set cmake args
+        components = dict()
+
+        components['aomp-extras'] = [
+            '../rocm-openmp-extras/aomp-extras',
+            '-DLLVM_DIR={0}'.format(llvm_prefix),
+            '-DDEVICE_LIBS_DIR={0}/amdgcn/bitcode'.format(devlibs_prefix),
+            '-DAOMP_STANDALONE_BUILD=0',
+            '-DDEVICELIBS_ROOT={0}'.format(devlibs_src),
+            '-DNEW_BC_PATH=1',
+            '-DAOMP={0}'.format(llvm_prefix),
+            '-DCMAKE_VERBOSE_MAKEFILE=1'
+        ]
+
+        # Shared cmake configuration for openmp, openmp-debug
+        openmp_common_args = [
+            '-DROCM_DIR={0}'.format(hsa_prefix),
+            '-DDEVICE_LIBS_DIR={0}/amdgcn/bitcode'.format(devlibs_prefix),
+            '-DAOMP_STANDALONE_BUILD=0',
+            '-DDEVICELIBS_ROOT={0}'.format(devlibs_src),
+            '-DOPENMP_TEST_C_COMPILER=$AOMP/bin/clang',
+            '-DOPENMP_TEST_CXX_COMPILER=$AOMP/bin/clang++',
```

Comment:
These won't expand to the value of `AOMP` like they would in bash, just so you know.